odl/deform/linearized.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators and functions for linearized deformation."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+from odl.discr import (DiscreteLp, DiscreteLpVector, Gradient, Divergence,
+                       PointwiseInner)
+from odl.operator.operator import Operator
+from odl.space import ProductSpaceVector
+
+
+__all__ = ('LinDeformFixedTempl', 'LinDeformFixedDisp')
+
+
+def _linear_deform(template, displacement, out=None):
+    """Linearized deformation of a template with a displacement field.
+
+    The function maps a given template ``I`` and a given displacement
+    field ``v`` to the new function ``x --> I(x + v(x))``.
+
+    Parameters
+    ----------
+    template : `DiscreteLpVector`
+        Template to be deformed by a displacement field.
+    displacement : `ProductSpace` of `DiscreteLp` element
+        The vector field (displacement field) used in the linearized
+        deformation.
+    out : `numpy.ndarray`, optional
+        Array to which the function values of the deformed template
+        are written. It must have the same shape as ``template`` and
+        a data type compatible with ``template.dtype``.
+
+    Returns
+    -------
+    deformed_template : `numpy.ndarray`
+        Function values of the deformed template. If ``out`` was given,
+        the returned object is a reference to it.
+
+    Examples
+    --------
+    We create a simple 1D template and a corresponding displacement
+    field. Where the displacement is zero, we expect that the output
```

Comment:
I don't agree. To me the sentence is more well-structured and readable with comma. It marks the end of a subordinate clause. Otherwise, readers have to fill this separation in their heads -- something I feel takes up energy and attention to other, more important things.